diff --git a/config/FindPythonLibs.cmake b/config/FindPythonLibs.cmake
index 1bbf3ce..51e8451 100644
--- a/config/FindPythonLibs.cmake
+++ b/config/FindPythonLibs.cmake
@@ -186,10 +186,11 @@ FOREACH(_CURRENT_VERSION ${_Python_VERSIONS})
     STRING(REGEX REPLACE "^#define[ \t]+PY_VERSION[ \t]+\"([^\"]+)\".*" "\\1"
                          _PYTHONLIBS_VERSION_STRING "${python_version_str}")
     UNSET(python_version_str)
+
+	# erase minor version
+	STRING(SUBSTRING ${_PYTHONLIBS_VERSION_STRING} 0 3 PYTHONLIBS_VERSION_STRING)
   ENDIF(PYTHON_INCLUDE_DIR AND EXISTS "${PYTHON_INCLUDE_DIR}/patchlevel.h")
 
-  # erase minor version
-  STRING(SUBSTRING ${_PYTHONLIBS_VERSION_STRING} 0 3 PYTHONLIBS_VERSION_STRING)
 
   IF (NOT ("${PYTHON_INCLUDE_DIR}" STREQUAL "${PYTHON_INCLUDE_DIR2}"))
 	  set(PYTHON_INCLUDE_DIR_TMP ${PYTHON_INCLUDE_DIR})
